Data/Vector/Mutable.hs

```diff
@@ -281,7 +281,12 @@ new :: PrimMonad m => Int -> m (MVector (PrimState m) a)
 {-# INLINE new #-}
 new = G.new
 
--- | Create a mutable vector of the given length. The memory is not initialized.
+-- | Create a mutable vector of the given length. The vector content
+--   should be presumed uninitialized. However exact semantics depends
+--   on vector implementation. For example unboxed and storable
+--   vectors will create vector filled with whatever underlying memory
+--   buffer happens to contain, while boxed vector initializes every
+--   element to @error "..."@.
```

Comment:
```suggestion
-- | Create a mutable vector of the given length. Boxed vector is
--  initialized with every element set to @error "..."@.
--
-- @since 0.5
```